Either/Either.swift

```diff
@@ -96,6 +96,38 @@ public func >>- <T, U, V> (either: Either<T, U>, @noescape transform: U -> Eithe
 }
 
 
+///  Extracts from a list of `Either` all the `Left` elements. All the `Left` elements are extracted in order.
+public func lefts <T, U>(list: [Either<T, U>]) -> [T] {
+	return list.reduce([T]()) { ( var memo, e) in
+		if let l = e.left {
+			memo.append(l)
+		}
+		
+		return memo
+	}
+}
+
+
+///  Extracts from a list of `Either` all the `Right` elements. All the `Right` elements are extracted in order.
+public func rights <T, U>(list: [Either<T, U>]) -> [U] {
+	return list.reduce([U]()) { ( var memo, e) in
+		if let l = e.right {
+			memo.append(l)
+		}
+		
+		return memo
+	}
+}
+
+
+///  Partitions a list of `Either` into two lists.
+///
+///  All the `Left` elements are extracted in order to the first component of the output. The `Right` elemests are extracted in order to the second component of the output.
+public func partitionEithers <T, U>(list: [Either<T, U>]) -> ([T], [U]) {
+	return (lefts(list), rights(list))
+}
```

Comment:
1. Since these all work on lists of Eithers instead of an actual Either, I'm not sure how that would work.

2. I tried that, but a) Had loads of trouble with the types. b) if using something like a `Result: EitherType` I'd rather have functions such as `errors`, `values`, and `partitionResults`

3. That does make sense, I'll take a look at it later.

4. Talk English dammit! But isn't this more a case of the input list being lazy rather than the functions themselves being lazy?